### PR TITLE
[Solutions] add e2e test for solution ui intergration

### LINKED ISSUE #1851: Add E2E testing of Solutions UI integration
**Component**: tests, solutions

**Why this is needed**: Prepare future integration to CI (in post-merge test suite), validate first iteration

**What should be done**: Implement a Cypress test suite for Environment creation and preparation from Platform UI (assume a Solution to be already deployed, wait for #1850 to test in Eve)

**Implementation proposal**:
`./node_modules/cypress/bin/cypress run -b chrome -s cypress/integration/Solutions.feature --no-exit`
- [x] Test creation of Environment (simple form)
- [x] Test addition of a Solution to an Environment (validate URL redirect)
- [x] Test deployment of a Solution after the preparation is complete
- [x] Test deletion of Environment

We want to do both tests (CLI & UI) on the same cluster

### PR DESCRIPTION
**Pull request by ChengYanJin**
**Component**: ui,solution, e2e test

**Context**: 
We need to implement the cypress test for the solution UI. To test the following scenarios:

- Test creation of Environment (simple form)
- Test addition of a Solution to an Environment (validate URL redirect)
- Test deployment of a Solution after the preparation is complete
- Test deletion of Environment

We should be able to run the different scenarios separately.

**Summary**:
Implement a Cypress test suite for Environment creation and preparation from Platform UI

**Acceptance criteria**: 
need to merge #1850 first so that the CI is ready.

- comment to execute the cypress test locally
`./node_modules/cypress/bin/cypress run -b chrome -s cypress/integration/solutions/CreateEnvironment.feature --no-exit`
`./node_modules/cypress/bin/cypress run -b chrome -s cypress/integration/solutions/AddSolutionToEnvironment.feature --no-exit`
`./node_modules/cypress/bin/cypress run -b chrome -s cypress/integration/solutions/DeleteEnvironment.feature --no-exit`

- Since we want to run the solution end to end test in CI in a separate VM to avoid flaky, we should be able to execute the solution frontend e2e separately by the following comment.
`npm run test:e2esolutions`

- In the CI, we should execute the solution e2e test by the following comment (Should use terraform to plug in the e2e test)
`npm run test:e2esolutions -- -e target_url="https://${CONTROL_PLANE_IP}:8443"`

Best Practice: Tests should always be able to be run independently from one another and still pass.
Anti-Pattern: Coupling multiple tests together.

---

Closes: #1851

### CODE REVIEW

File: ui/cypress/integration/Solutions.feature [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Feature: Solutions 
+   Scenario: Environment List
+     Given I log in
+     When I go to the solutions list by clicking the solution icon in the sidebar
+     And I go to the Create Envrionments page by clicking the Create a New Environment button
```
Comment (alexandre-allard) [inline ui/cypress/integration/Solutions.feature]:
```suggestion
     And I go to the Create Environments page by clicking the Create a New Environment button
```

File: ui/cypress/integration/common/solutions.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      requestTimeout: 60000,
+      responseTimeout: 60000,
+    };
+    cy.wait('@getEnvironmentsList', timeOut);
+    cy.get('.sc-table:first .sc-table-row').should('have.length', 1); //Only Table header
+  },
+);
+
+And(
+  'I go to the Create Envrionments page by clicking the Create a New Environment button',
```
Comment (alexandre-allard) [inline ui/cypress/integration/common/solutions.js]:
```suggestion
  'I go to the Create Environments page by clicking the Create a New Environment button',
```

File: ui/cypress/integration/common/solutions.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  () => {
+    cy.get('[data-cy="create_new_environment_button"]').click();
+  },
+);
+
+Then(
+  'I fill out the Create Envrionments form and check if the environment I created is displayed on the Environments list',
```
Comment (alexandre-allard) [inline ui/cypress/integration/common/solutions.js]:
```suggestion
  'I fill out the Create Environments form and check if the environment I created is displayed on the Environments list',
```

File: ui/cypress/integration/Solutions.feature (hunk 1 above)
Comment (ChengYanJin) [reply ui/cypress/integration/Solutions.feature]: Sorry for the spelling mistakes.
And thank you for the correction!

**Review by MonPote — APPROVED** [review]
LGTM

**Pull request merged**